src/javaRosa.js

```diff
@@ -1834,6 +1834,9 @@ define([
                     if (hasText && !value.autoId) {
                         data[name] = value.id;
                     }
+                    if (value.hasMarkdown) {
+                        data[name + ':hasMarkdown'] = value.hasMarkdown;
```

Comment:
I imagine `value.hasMarkdown` is normally a boolean value, which means this code will be fine. Is there any possibility that it might not be a boolean value but still evaluate to `true`? If yes, do you think it would be worth changing this line to `data[name + ':hasMarkdown'] = true;`?